src/marker.tsx

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import ProjectedLayer from './projected-layer';
+
+export interface Props {
+  coordinates: number[];
+  anchor?: any;
+  offset?: any;
+  children?: JSX.Element;
+  onClick?: React.MouseEventHandler<HTMLDivElement>;
+  onMouseEnter?: React.MouseEventHandler<HTMLDivElement>;
+  onMouseLeave?: React.MouseEventHandler<HTMLDivElement>;
+  style?: React.CSSProperties;
+}
+
+export default class Marker extends React.Component<Props, void> {
+  public render() {
+    return (
+      <ProjectedLayer
+        {...{ ...this.props, children: undefined}}
```

Comment:
This isn't nice to read... Also it's unnecessary. 1) It might not compile to just `createElement(ProjectedLayer, this.props)`, but if it does, that'd be fine, because immutability is guaranteed on these objects.